pkg/web/apps/apps_psql_repository.go

```diff
@@ -186,24 +311,6 @@ func (a *appsPostgreSQLRepository) CreateApp(id, appId, name string) error {
 	return nil
 }
 
-func (a *appsPostgreSQLRepository) GetAppByAppID(appID string) (*models.App, error) {
```

Comment:
Hi @craicoverflow, 

Before change or remove a method is required check if it has been or not used in the project. In this case, it is used in the service binding and cannot be removed or changed in order to filter and return just the active ones. There is required to get the app by app id when it is disabled too. 

To return apps which are just active the project has already a method implemented which is GetActiveAppID the is also a good idea follow the same nomenclature. I recommend you create a method called GetActiveAppByAppID and keep this one.

Also, note that when a dev will use a method GetObjectByID is not intuitive at all expected that it has been filtered by any other thing that is not the appID. It is important to highlight that indeed the init will use fetch the disabled apps too since it will do the binding steps in the future. Is it make sense?

Reply:
> Hi @craicoverflow,
> 
> In this case, it is used in the service [binding](https://github.com/aerogear/mobile-security-service/blob/a0aae35c13836f3250a1190707797e7072d0ca6f/pkg/web/apps/apps_service.go#L97) and cannot be removed or changed in order to filter and return just the active ones. There is required to get the app by app id when it is disabled too.
> 
> To return apps which are just active the project has already a method implemented which is GetActiveAppID then, in my point of view, is also a good idea follow the same nomenclature. I recommend you create a method called GetActiveAppByAppID and keep this one.
> 
> Also, note that when a dev will use a method GetObjectByID is not intuitive at all expected that it will be filtering by any other thing that is not the appID, do not you agree?. It is important to highlight that indeed the /init will fetch the disabled apps too in the neext steps when it will be creating/updating apps, I mean doing the binding. Is it make sense? Could you please do this fix before merge?

Hi @camilamacedo86 

I will do the change in order to remain consistent but I disagree with the sentiment. 

They are **deleted** not **inactive**. It should be assumed that an endpoint will never return deleted rows from the database, it should not need to be specified in the function name.

And if we are going the way of having appropriate naming then it would not be `Get**Active**AppByAppID` because at no stage are the apps ever active or inactive.

Reply:
Hi @craicoverflow,

Following some considerations:
* The name delete was a not good choice since the app is NOT DELETED at all as its versions and device.
* If an App is updated as "deleted_at" it should not return in the search, however, it can be undeleted again and according to all definitions, its versions and devices should come back to appear. So, it is more active and inactive app. 
* The endpoint is one layer and the repository methods are another. If you are implementing a service in the REST you will never suppose that a method with the name GetAppByID will return just the active ones or where the deleted_at is not null or not "soft deleted" as you can choose the name.

Reply:
> Hi @craicoverflow,
> 
> Following some considerations:
> 
> * The name delete was IHMO a not good choice at all since the app is NOT DELETED as its versions and device and never will be.
> * If an App has an updated as "deleted_at" it should not return in the search, however, it can be undeleted again and according to all definitions, its versions and devices should come back to appear too. So, according to the requirements, it is more like to disable an app then delete, do not you agree?
> * The endpoint is one layer and the repository methods are another. If you are implementing a service in the REST you will never assume that a method with the name GetAppByID will return just the active ones or where the deleted_at is not null or not "soft deleted" as you can prefer the nomenclature.
> * Before remove or change the implementation of any method in any project is mandatory to check if it has or not been used. By the IDE you can check where something is or not used easily.  Is it make sense? We cannot change the implementation of a method that has been used for other services.

Hi @camilamacedo86

To help me understand your considerations, where in the requirements does it say that apps can or will be undeleted?

Reply:
Hi @camilamacedo86 I have updated the function name to be consistent with the rest of the server.

Reply:
Hi @craicoverflow,

Following the answers. 

1. To help me understand your considerations, where in the requirements does it say that apps can or will be undeleted?

* Actions/Task: The unbinding and binding services. 
* You can check the comments in the JIRA to know how it was defined. 

2. According to your considerations, do you also think that the endpoint URL should be modified to include this information?

NO. The endpoint should not be changed. It is required just to keep the method in the repo which has been used for the binding service.